kahypar/io/hypergraph_io.h

```diff
@@ -77,22 +146,22 @@ static inline void readHypergraphFile(const std::string& filename, HypernodeID&
                                        hypergraph_type == HypergraphType::EdgeAndNodeWeights ?
                                        true : false;
 
+    std::vector<size_t> line_number_vector;
     index_vector.reserve(static_cast<size_t>(num_hyperedges) +  /*sentinel*/ 1);
     index_vector.push_back(edge_vector.size());
+    if (line_number_vector != nullptr) {
```

Comment:
`line_number_vector == nullptr` can never happen since you create the vector 3 lines above, right?

Reply:
Did you mean to check `if (validate_input)`?